core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java

```diff
@@ -499,6 +508,26 @@ public void write(final String name, final byte[] data) {
                 // custom run time converters
                 carc.invokeStaticMethod(CS_POPULATE_CONVERTERS, builder);
 
+                // cache explicit converts and make them available during runtime
```

Comment:
@dmlloyd I opted for this approach. WDYT?

Reply:
This looks better, thanks!

Reply:
Thanks for the initial suggestion :-)